clients/drcachesim/scheduler/scheduler_impl.h

```diff
@@ -570,6 +554,21 @@ template <typename RecordType, typename ReaderType> class scheduler_impl_tmpl_t
     virtual scheduler_status_t
     set_initial_schedule(std::unordered_map<int, std::vector<int>> &workload2inputs) = 0;
 
+    // When an output's input changes from a valid (not INVALID_INPUT_ORDINAL) input to
```

Comment:
Such changes start to turn it into a single call passing both the prev and new: which is harder to support interleaved with set_cur_input() operations at the right time.  But if it remains only called on a change, maybe it still works.  Originally it was simpler at the call site to not allow a previous inactive (with locking assumptions); but now it could be moved.  OK, done.